packages/clarity-js/src/data/metadata.ts

```diff
@@ -99,15 +109,19 @@ function num(string: string, base: number = 36): number {
 }
 
 function user(): number {
-  let id;
+  let id = cookie(Constant.STORAGE_KEY);
+  return id && id.length === 8 ? num(id) : num(shortid());
+}
+
+function cookie(key: string): string {
   let cookies: string[] = document.cookie.split(Constant.SEMICOLON);
   if (cookies) {
     for (let i = 0; i < cookies.length; i++) {
       let pair: string[] = cookies[i].split(Constant.EQUALS);
-      if (pair.length > 1 && pair[0].indexOf(Constant.STORAGE_KEY) >= 0 && pair[1].length === 8) {
-        id = pair[1];
+      if (pair.length > 1 && pair[0].indexOf(key) >= 0) {
```

Comment:
Why match on just the partial string for key?

Reply:
Fixed it to match full string.